doc/scripts/spicy-doc-to-rst

```diff
@@ -296,6 +296,43 @@ class Method:
         return self.rst() < other.rst()
 
 
+# Manages a single output file.
+class OutputFile:
+    def __init__(self, ns, args):
+        self.fd = sys.stdout
+        self.fname = None
+        self.fname_tmp = None
+
+        if args.dir:
+            self.fname = namespace(ns)
+
+            if self.fname.endswith("_"):
+                self.fname = self.fname[:-1]
+
+            self.fname = (
+                self.fname.lower().replace("::", "-").replace("_", "-") + ".rst"
+            )
+            self.fname = os.path.join(args.dir, self.fname)
+            self.fname_tmp = self.fname + ".tmp"
+            self.fd = open(self.fname_tmp, "w+")
+        else:
+            self.fd = sys.stdout
+
+    def close(self):
```

Comment:
Let's instead implement [`__del__`](https://docs.python.org/3.9/reference/datamodel.html#object.__del__) so user do not have to worry about manual lifetime management,

```suggestion
    def __del__(self):
```

Reply:
We actually need the manual close as we could open the same file multiple times, and need the previous one to be closed again already, which with `__del__` I don't think we could guarantee.